dirty_cat/test/test_minhash_encoder.py

```diff
@@ -0,0 +1,91 @@
+import numpy as np
+import time
+
+from sklearn.datasets import fetch_20newsgroups
+from dirty_cat.minhash_encoder import MinHashEncoder
+
+def test_MinHashEncoder(n_sample=70, minmax_hash=False):
+    X_txt = fetch_20newsgroups(subset='train')['data']
+    X = X_txt[:n_sample]
+
+    for minmax_hash in [True, False]:
+        for hashing in ['fast', 'murmur']:
+
+            if minmax_hash and hashing == 'murmur':
+                pass # not implemented
+
+            # Test output shape
+            encoder = MinHashEncoder(n_components=50, hashing=hashing)
+            encoder.fit(X)
+            y = encoder.transform(X)
+            assert y.shape == (n_sample, 50), str(y.shape)
+            assert len(set(y[0])) == 50
+
+            # Test same seed return the same output
+            encoder = MinHashEncoder(50, hashing=hashing)
+            encoder.fit(X)
+            y2 = encoder.transform(X)
+            np.testing.assert_array_equal(y, y2)
+
+            # Test min property
+            if not minmax_hash:
+                X_substring = [x[:x.find(' ')] for x in X]
+                encoder = MinHashEncoder(50, hashing=hashing)
+                encoder.fit(X_substring)
+                y_substring = encoder.transform(X_substring)
+                np.testing.assert_array_less(y - y_substring, 0.0001)
+
+def test_input_type():
+    # Numpy array
+    X = np.array(['alice', 'bob'])
+    enc = MinHashEncoder(n_components=2)
+    enc.fit_transform(X)
+    # List
+    X = ['alice', 'bob']
+    enc = MinHashEncoder(n_components=2)
+    enc.fit_transform(X)
+
+def test_get_unique_ngrams():
+    string = 'test'
+    true_ngrams = {
+        (' ','t'), ('t','e'), ('e','s'), ('s', 't'),
+        ('t',' '), (' ','t','e'), ('t','e','s'),
+        ('e','s','t'), ('s','t',' '), (' ','t','e','s'),
+        ('t','e','s','t'), ('e','s','t',' ')}
+    ngram_range = (2,4)
+    enc = MinHashEncoder(n_components=2)
+    ngrams = enc.get_unique_ngrams(string, ngram_range)
+    assert ngrams == true_ngrams
+        
+def profile_encoder(Encoder, hashing='fast', minmax_hash=False):
+
+    from dirty_cat import datasets
+    employee_salaries = datasets.fetch_employee_salaries()
+    data = pd.read_csv(employee_salaries['path'])
```

Comment:
This function computes the time needed to encode a dataset, but rely on pandas to load it.

Reply:
Yes indeed. 
It was only to assert performances. 
It do not run with unit test, so we could keep it (or remove it).